training/training/core/dataset.py

```diff
@@ -98,3 +101,75 @@ def getCategoryList(self) -> list[str]:
         if self._category_list is None:
             raise Exception("Category list not available")
         return self._category_list
+
+
+class DefaultImageDatasets(Enum):
+    MNIST = "MNIST"
+    FASHION_MNIST = "FashionMNIST"
+    KMNIST = "KMNIST"
+
+
+class ImageDefaultDatasetCreator(TrainTestDatasetCreator):
+    def __init__(
+        self,
+        dataset_name: str,
+        train_transform: None,
+        test_transform: None,
+        batch_size: int = 32,
+        shuffle: bool = True,
+    ):
+        if dataset_name not in DefaultImageDatasets.__members__:
+            raise Exception(
+                f"The {dataset_name} file does not currently exist in our inventory. Please submit a request to the contributors of the repository"
+            )
+        self.train_transform = train_transform or transforms.Compose(
+            [transforms.toTensor()]
+        )
+        self.test_transform = test_transform or transforms.Compose(
+            [transforms.toTensor()]
+        )
+        self.batch_size = batch_size
+        self.shuffle = shuffle
+
+        self.train_set = datasets.__dict__[dataset_name](
+            root="./backend/image_data_uploads",
```

Comment:
please use ./training/image_data_uploads but ensure that after dataloader created the downloaded dataset deleted. might wanna confirm with torch docs here @codingwithsurya

Reply:
Ok will create that directory. And u would want a similar structure as in backend/image_data_uploads? right? but obviously in that django blueprint.

Reply:
yes

Reply:
I've DM'd you on discord this detail

Reply:
sure, check out this gist on what im thinking we can do:

https://gist.github.com/codingwithsurya/950097788ac8f60b561d62ca087fed5f

Reply:
we discussed over dm on this @codingwithsurya

Reply:
implemented. Testing will confirm if this truly works.